CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,67 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array &$params) {
+    $reviewFieldsLegalOptions = [
+      'id' => FILTER_VALIDATE_INT,
+      'weight' => FILTER_VALIDATE_INT,
+      'required' => FILTER_VALIDATE_BOOLEAN
+    ];
+    $reviewFields = CRM_Utils_Array::value('review_fields', $params);
+    if (empty($reviewFields)) {
+      return;
+    }
+
+    $filteredReviewFields = [];
+    foreach ($reviewFields as $reviewFieldOptions) {
+      $invalidReviewFieldOptions = [];
+      $validReviewFieldOptions = array_filter(
+        $reviewFieldOptions,
+        function($value, $key) use ($reviewFieldsLegalOptions, &$invalidReviewFieldOptions) {
+          if (!isset($reviewFieldsLegalOptions[$key])) {
+            return FALSE;
+          }
+
+          if (filter_var($value, $reviewFieldsLegalOptions[$key], FILTER_NULL_ON_FAILURE) !== NULL) {
```

Comment:
I have some reservations about suing the `FILTER_NULL_ON_FAILURE` flag because if you check the manual here: https://www.php.net/manual/en/filter.filters.validate.php, the filer is valid for `FILTER_VALIDATE_BOOLEAN` only and not for `FILTER_VALIDATE_INT`, so we should find an alternative.